src/terminal/adapter/adaptDispatch.cpp

```diff
@@ -1578,64 +1522,47 @@ bool AdaptDispatch::_EraseScrollback()
     if (fSuccess)
     {
         const SMALL_RECT Screen = csbiex.srWindow;
-        const short sWidth = Screen.Right - Screen.Left;
         const short sHeight = Screen.Bottom - Screen.Top;
-        FAIL_FAST_IF(!(sWidth > 0 && sHeight > 0));
+        FAIL_FAST_IF(!(sHeight > 0));
         const COORD Cursor = csbiex.dwCursorPosition;
 
         // Rectangle to cut out of the existing buffer
+        // It will be clipped to the buffer boundaries so SHORT_MAX gives us the full buffer width.
         SMALL_RECT srScroll = Screen;
+        srScroll.Left = 0;
+        srScroll.Right = SHORT_MAX;
         // Paste coordinate for cut text above
         COORD coordDestination;
         coordDestination.X = 0;
         coordDestination.Y = 0;
 
-        // Fill character for remaining space left behind by "cut" operation (or for fill if we "cut" the entire line)
-        CHAR_INFO ciFill;
-        ciFill.Attributes = csbiex.wAttributes;
-        ciFill.Char.UnicodeChar = static_cast<WCHAR>(0x20); // space character. use 0x20 instead of literal space because we can't assume the compiler will always turn ' ' into 0x20.
-        fSuccess = !!_conApi->ScrollConsoleScreenBufferW(&srScroll, nullptr, coordDestination, &ciFill);
+        // Typically a scroll operation should fill with standard erase attributes, but in
+        // this case we need to use the default attributes, hence standardFillAttrs is false.
+        fSuccess = !!_conApi->PrivateScrollRegion(srScroll, std::nullopt, coordDestination, false);
         if (fSuccess)
         {
-            // Clear everything after the viewport. This is two regions:
-            // A. below the viewport
-            // B. to the right of the viewport.
-
-            // First clear section A
+            // Clear everything after the viewport.
             const DWORD dwTotalAreaBelow = csbiex.dwSize.X * (csbiex.dwSize.Y - sHeight);
             const COORD coordBelowStartPosition = { 0, sHeight };
-            // We don't use the _EraseAreaHelper here because _EraseSingleLineDistanceHelper does it all in one operation
-            fSuccess = _EraseSingleLineDistanceHelper(coordBelowStartPosition, dwTotalAreaBelow, csbiex.wAttributes);
+            // Again we need to use the default attributes, hence standardFillAttrs is false.
+            fSuccess = _conApi->PrivateFillRegion(coordBelowStartPosition, dwTotalAreaBelow, L' ', false);
```

Comment:
nit

```suggestion
            fSuccess = _conApi->PrivateFillRegion(coordBelowStartPosition, dwTotalAreaBelow, UNICODE_SPACE, false);
```

if we can see UNICODE_SPACE here?